auto-value-gson/src/test/java/com/ryanharter/auto/value/gson/AutoValueGsonExtensionTest.java

```diff
@@ -548,4 +547,125 @@ public void simple() {
       .and()
       .generatesSources(expected);
   }
+
+  @Test public void handlesGenericTypes() {
+    JavaFileObject source1 = JavaFileObjects.forSourceString("test.Foo", ""
+        + "package test;\n"
+        + "import com.google.auto.value.AutoValue;\n"
+        + "import com.google.gson.Gson;\n"
+        + "import com.google.gson.TypeAdapter;\n"
+        + "import com.google.gson.reflect.TypeToken;\n"
+        + "@AutoValue public abstract class Foo<A, B, C> {\n"
+        + "  public static TypeAdapter<Foo> typeAdapter(Gson gson, TypeToken<? extends Foo> typeToken) {\n"
+        + "    return new AutoValue_Foo.GsonTypeAdapter(gson, typeToken);"
+        + "  }\n"
+        + "  public abstract C c();\n"
+        + "  public abstract A a();\n"
+        + "  public abstract B b();\n"
+        + "  public abstract String d();\n"
+        + "}"
+    );
+
+    JavaFileObject expected = JavaFileObjects.forSourceString("test.AutoValue_Test", ""
+        + "package test;\n"
+        + "\n"
+        + "import com.google.gson.Gson;\n"
+        + "import com.google.gson.TypeAdapter;\n"
+        + "import com.google.gson.reflect.TypeToken;\n"
+        + "import com.google.gson.stream.JsonReader;\n"
+        + "import com.google.gson.stream.JsonToken;\n"
+        + "import com.google.gson.stream.JsonWriter;\n"
+        + "import java.io.IOException;\n"
+        + "import java.lang.Object;\n"
+        + "import java.lang.Override;\n"
+        + "import java.lang.String;\n"
+        + "import java.lang.reflect.ParameterizedType;\n"
+        + "import java.lang.reflect.Type;\n"
+        + "\n"
+        + "final class AutoValue_Foo<A, B, C> extends $AutoValue_Foo<A, B, C> {\n"
+        + "  AutoValue_Foo(C c, A a, B b, String d) {\n"
+        + "    super(c, a, b, d);\n"
+        + "  }\n"
+        + "\n"
+        + "  public static final class GsonTypeAdapter extends TypeAdapter<Foo> {\n"
```

Comment:
Actually it looks like there is a `TypeAdapterFactory` generated, it's just omitted from the tests.

I think this could be fixed then by changing
```java
public static TypeAdapter<Foo> typeAdapter(Gson gson, TypeToken<? extends Foo> typeToken)
```
to
```java
public static <A, B, C> TypeAdapter<Foo<A, B, C>> typeAdapter(Gson gson, TypeToken<? extends Foo<A, B, C>> typeToken)
```
and then properly hoisting the generics onto `GsonTypeAdapter`.